presto-hive/src/main/java/com/facebook/presto/hive/orc/DwrfSelectivePageSourceFactory.java

```diff
@@ -83,6 +90,10 @@ public DwrfSelectivePageSourceFactory(
         this.orcFileTailSource = requireNonNull(orcFileTailSource, "orcFileTailSource is null");
         this.stripeMetadataSource = requireNonNull(stripeMetadataSource, "stripeMetadataSource is null");
         this.fileOpener = requireNonNull(fileOpener, "fileOpener is null");
+        tupleDomainFilterCache = CacheBuilder.newBuilder()
```

Comment:
- Shall we inject the cache as a singleton instead of having two for orc and dwrf? An example can be found in `createOrcFileTailSource`.
- A `CacheStatsMBean` would be great if we are to call `recordStats()`.

Reply:
@highker Good point. Let me do this.